install/kubernetes/helm/subcharts/prometheus/templates/deployment.yaml

```diff
@@ -30,6 +30,14 @@ spec:
 {{- if .Values.global.priorityClassName }}
       priorityClassName: "{{ .Values.global.priorityClassName }}"
 {{- end }}
+      initContainers:
+      - name: prom-init
+        image: "busybox"
+        command: ['sh', '-c', 'counter=0; until [ -d /etc/istio-certs ] || [ "$counter" -ge 30 ]; do echo waiting for istio certs; sleep 1; counter=$((counter+1)); done;']
```

Comment:
so, this is a ~30s timeout? on start of this container?  have we observed that certs are typically ready within 30s with installs?

Reply:
Yes, on start of prom. Based on my experience certs will definitely be ready within 30s. It is actually quite sufficient timeout.

Reply:
This will wait for 30s and then continue with initialization. 
We want init-container to fail if certs are not ready.

Reply:
Then it will make citadel a dependence for prometheus. Is this ideal?

Reply:
I think that is fine.

Reply:
If we have configured certs to be there, then we should wait for it.

Reply:
Done, ptal.